src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.cs

```diff
@@ -130,5 +148,64 @@ private void OnErrorListPropertyChanged(object sender, PropertyChangedEventArgs
                 AddTableSourceIfNecessary(this.Workspace.CurrentSolution);
             }
         }
+
+        private void OnErrorListFullSolutionAnalysisToggled(object sender, AnalysisToggleStateChangedEventArgs e)
+        {
+            var newOptions = _optionService.GetOptions()
+                .WithChangedOption(RuntimeOptions.FullSolutionAnalysis, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.CSharp, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.VisualBasic, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, TypeScriptLanguageName, e.NewState);
+            
+            _optionService.SetOptions(newOptions);
+        }
+
+        private void OnOptionChanged(object sender, OptionChangedEventArgs e)
+        {
+            Contract.ThrowIfFalse(_errorList is IErrorList2);
+
+            if (e.Option == RuntimeOptions.FullSolutionAnalysis || e.Option == ServiceFeatureOnOffOptions.ClosedFileDiagnostic)
+            {
+                var analysisDisabled = !(bool)e.Value;
+                if (analysisDisabled)
+                {
+                    ((IErrorList2)_errorList).AnalysisToggleState = false;
+                }
+            }
+        }
+
+        internal static bool ErrorListHasFullSolutionAnalysisButton()
+        {
+            try
+            {
+                // Full solution analysis option has been moved to the error list from Dev14 Update3.
+                // Use reflection to check if the new interface "IErrorList2" exists in Microsoft.VisualStudio.Shell.XX.0.dll.
+                return Assembly.GetAssembly(typeof(ErrorHandler)).GetType("Microsoft.Internal.VisualStudio.Shell.IErrorList2") != null;
+            }
+            catch (Exception)
+            {
+                // Ignore exceptions.
+                return false;
+            }
+        }
+
+        // TODO: Remove the below temporary types 'IErrorList2' and 'AnalysisToggleButtonStateChangedEventArgs' when we move to new version of Microsoft.VisualStudio.Shell.XX.0.dll that has these types.
+        private interface IErrorList2
```

Comment:
This types were added just so project builds. This won't work until we actually compile against newer shell binary, at which point we only need to delete these types and it should work (just minimizing potential ask mode changes)